components/infobox/wikis/counterstrike/infobox_league_custom.lua

```diff
@@ -34,46 +38,99 @@ local _MODE_TEAM = 'team'
 local _ICON_EPT_CHALLENGER = '[[File:ESL Pro Tour Challenger.png|20x20px|link='
 local _ICON_EPT_MASTERS = '[[File:ESL Pro Tour Masters.png|20x20px|link='
 
+local _league
+
 function CustomLeague.run(frame)
 	local league = League(frame)
-	league.addCustomCells = CustomLeague.addCustomCells
-	league.createTier = CustomLeague.createTier
-	league.createPrizepool = CustomLeague.createPrizepool
-	league.addCustomContent = CustomLeague.addCustomContent
+	_league = league
+	league.createWidgetInjector = CustomLeague.createWidgetInjector
 	league.defineCustomPageVariables = CustomLeague.defineCustomPageVariables
 	league.addToLpdb = CustomLeague.addToLpdb
 	league.getWikiCategories = CustomLeague.getWikiCategories
 
 	return league:createInfobox(frame)
 end
 
-function CustomLeague:addCustomCells(infobox, args)
-	infobox:cell('Game', CustomLeague:_createGameCell(args))
-	infobox:cell('Teams', (args.team_number or '') .. (args.team_slots and ('/' .. args.team_slots) or ''))
-	infobox:cell('Players', args.player_number)
-	infobox:fcell(
-		Cell:new('[[File:ESL 2019 icon.png|20x20px|link=|ESL|alt=ESL]] Pro Tour Tier')
-			:content(CustomLeague:_createEslProTierCell(args.eslprotier))
-			:categories(
-				function(_, ...)
-					infobox:categories('ESL Pro Tour Tournaments')
-				end
-			)
-			:make()
-	)
-	infobox:fcell(
-		Cell:new(Template.safeExpand(mw.getCurrentFrame(), 'Valve/infobox'))
-			:addClass('valvepremier-highlighted')
-			:content(CustomLeague:_createValveTierCell(args.valvetier))
-			:categories(
-				function(_, ...)
-					infobox:categories('Valve Sponsored Tournaments')
-				end
-			)
-			:make()
-	)
+function CustomLeague:createWidgetInjector()
+	return CustomInjector()
+end
+
+function CustomInjector:addCustomCells(widgets)
+	local args = _league.args
+	table.insert(widgets, Cell{
+		name = 'Teams',
+		content = {(args.team_number or '') .. (args.team_slots and ('/' .. args.team_slots) or '')}
+	})
+	table.insert(widgets, Cell{
+		name = 'Game',
+		content = {CustomLeague:_createGameCell(args)}
+	})
```

Comment:
could maybe be moved into the `Customizable` with id `server`

Reply:
I don't see how, it's a different thing